modules/gapi/include/opencv2/gapi/imgproc.hpp

```diff
@@ -222,6 +222,32 @@ namespace imgproc {
         }
     };
 
+    G_TYPED_KERNEL(GYUV2Gray, <GMat(GMat)>, "org.opencv.colorconvert.imgproc.yuvtogray") {
+        static GMatDesc outMeta(GMatDesc in) {
+            GAPI_Assert(in.depth  == CV_8U);
+            GAPI_Assert(in.planar == false);
+            GAPI_Assert(in.size.width  % 2 == 0);
+            GAPI_Assert(in.size.height % 3 == 0);
+
+            return {CV_8U, 1, Size{in.size.width, in.size.height - (in.size.height / 3)}, false};
+        }
+    };
```

Comment:
I am not sure if we need to have this very specific kernel defined in a public header -- since it serves as an implementation detail only

Reply:
I don't get why width must be odd and height must be even. Why?

Reply:
So now I've got that it is not a building block of a compound kernel but in fact an I420 conversion. Please rename and comment it appropriately as I stated in the test comments

Reply:
oh no, it is still the building block for NV12Gray. So then you don't need it in public here.

Reply:
Fixed